pkg/client/optimizely_user_context.go

```diff
@@ -46,17 +49,17 @@ func newOptimizelyUserContext(optimizely *OptimizelyClient, userID string, attri
 }
 
 // GetOptimizely returns optimizely client instance for Optimizely user context
-func (o *OptimizelyUserContext) GetOptimizely() *OptimizelyClient {
+func (o OptimizelyUserContext) GetOptimizely() *OptimizelyClient {
 	return o.optimizely
 }
 
 // GetUserID returns userID for Optimizely user context
-func (o *OptimizelyUserContext) GetUserID() string {
+func (o OptimizelyUserContext) GetUserID() string {
 	return o.userID
 }
 
 // GetUserAttributes returns user attributes for Optimizely user context
-func (o *OptimizelyUserContext) GetUserAttributes() map[string]interface{} {
+func (o OptimizelyUserContext) GetUserAttributes() map[string]interface{} {
```

Comment:
why do we need a copy of user attributes ? are we allowed to modify them?

Reply:
The reason for returning a copy of the attributes is to not allow user to make changes to the stored attributes directly by calling `GetUserAttributes` and making changes to it since `map` is a pointer type. To test this behaviour, a test can be seen here:
https://github.com/optimizely/go-sdk/pull/299/files#diff-97b3b48891edc303a7563bdf3ec2adfc96cc4819bdca644bc277d29a78d55d2cR75